Blocks community events card (#27)
huggingface/community-events: gradio-blocks/README.md

# Gradio Blocks Party  🥳

> _**Timeline**: May 11th, 2022 - May 31st, 2022_

Reviewer: To update, I think it was moved a couple of days :smile:

Author: right, I think the internal launch for HF is still today, will update the public launch date to Monday

Author: updated

---

> Happy to invite you to the Gradio Blocks Party - a community event in which we will setup spaces for models using the new Gradio Blocks feature. Blocks allows you to build web-based demos in a flexible way using the Gradio library. The event will take place from May 11th to 31st. We will be organizing this event on github and the huggingface discord channel.

Reviewer 2: ```suggestion
Happy to invite you to the Gradio Blocks Party - a community event in which we will build cool machine learning demos using the new Gradio Blocks feature. Blocks allows you to build web-based demos in a flexible way using the Gradio library. The event will take place from May 11th to 31st. We will be organizing this event on github and the huggingface discord channel.
```

**What is Gradio?**

Gradio is a Python library that allows you to quickly build web-based machine learning demos, data science dashboards, or other kinds of web apps, entirely in Python. These web apps can be launched from wherever you use Python (jupyter notebooks, colab notebooks, Python terminal, etc.) and shared with anyone instantly using Gradio's auto-generated share links. To learn more about Gradio see the Get Started Guide: https://gradio.app/getting_started/

Gradio can be installed via pip

```pip install gradio```

**What is Blocks?**

> gradio.Blocks is a low-level API that allows you to have full control over the data flows and layout of your application. You can build very complex, multi-step applications using Blocks. If you have already used gradio.Interface, you know that you can easily create fully-fledged machine learning demos with just a few lines of code. The Interface API is very convenient but in some cases may not be sufficiently flexible for your needs. For example, you might want to:

Reviewer 2: ```suggestion
`gradio.Blocks` is a low-level API that allows you to have full control over the data flows and layout of your application. You can build very complex, multi-step applications using Blocks. If you have already used gradio.Interface, you know that you can easily create fully-fledged machine learning demos with just a few lines of code. The Interface API is very convenient but in some cases may not be sufficiently flexible for your needs. For example, you might want to:
```

> Group together related demos as multiple tabs in one web app
> Change the layout of your demo instead of just having all of the inputs on the left and outputs on the right
> Have multi-step interfaces, in which the output of one model becomes the input to the next model, or have more flexible data flows in general
> Change a component's properties (for example, the choices in a Dropdown) or its visibilty based on user input

Reviewer: ```suggestion
* Group together related demos as multiple tabs in one web app.
* Change the layout of your demo instead of just having all of the inputs on the left and outputs on the right.
* Have multi-step interfaces, in which the output of one model becomes the input to the next model, or have more flexible data flows in general.
* Change a component's properties (for example, the choices in a Dropdown) or its visibility based on user input.
```

Author: update thanks

> to learn more about Blocks see the guide https://www.gradio.app/introduction_to_blocks/

Reviewer: ```suggestion
To learn more about Blocks, see the [official guide](https://www.gradio.app/introduction_to_blocks/).
```

> **How Does Gradio and Hugging Face work together?**

Reviewer 2: ```suggestion
**How Do Gradio and Hugging Face work together?**
```

Hugging Face Spaces is a free hosting option for Gradio demos. Spaces comes with 3 SDK options: Gradio, Streamlit and Static HTML demos. Spaces can be public or private and the workflow is similar to github repos. There are over 2000+ Gradio spaces currently on Hugging Face. Learn more about spaces [here](https://huggingface.co/docs/hub/spaces).

> **Event Plan**

Reviewer 2: I feel like we should make this more concrete so that people who read the competition have a very clear idea of what this event is. Like it should be a 1-2-3-4-5
1. learn about blocks (maybe browse some example blocks)
2. build your own blocks
3. submit to the competition org
4. share the word and accumulate likes
5. win prizes

Author: updated

> main components of the event consist of:

Reviewer 3: ```suggestion
In this event, you will:
```

- How Gradio and Hugging Face Spaces work together
- Understanding the Hugging Face ecosystem
- Share Demos in a Hugging Face organization for the Blocks event

**Example spaces using Blocks**
- [https://huggingface.co/spaces/akhaliq/ArcaneGAN-blocks](https://huggingface.co/spaces/akhaliq/ArcaneGAN-blocks)
- [https://huggingface.co/spaces/merve/gr-blocks](https://huggingface.co/spaces/merve/gr-blocks)
- [https://huggingface.co/spaces/osanseviero/tortoisse-tts](https://huggingface.co/spaces/osanseviero/tortoisse-tts)
- [https://huggingface.co/spaces/akhaliq/CaptchaCracker](https://huggingface.co/spaces/akhaliq/CaptchaCracker)

To participate in the event

- Join the organization for Blocks event
    - [https://huggingface.co/Gradio-Blocks](https://huggingface.co/Gradio-Blocks)
- Join the discord
    - [discord](https://discord.com/invite/feTf9x3ZSB)

Participants will have a month to build and share Gradio demos using the Blocks feature. We will share a list of ideas of spaces that can be created using blocks or participants are free to try out their own ideas. At the end of the event, spaces will be evaluated for creativity and prizes will be given 

> potential ideas for creating spaces:

Reviewer 2: We should make it clear that they don't have to restrict themselves to these

Author: sounds good, ill update to add that

Author: updated

- papers from https://paperswithcode.com/
- themed spaces: Vision, NLP, Audio based spaces
- Models from huggingface model hub
- Models from other model hubs
    - Tensorflow Hub: see example Gradio demos at https://huggingface.co/tensorflow
    - Pytorch Hub: see example Gradio demos at https://huggingface.co/pytorch
    - ONNX model Hub: see example Gradio demos at https://huggingface.co/onnx
    - PaddlePaddle Model Hub: see example Gradio demos at https://huggingface.co/PaddlePaddle
- participant ideas

> **Prizes (tentative list)**

Reviewer 2: Let's finalize this list asap

Author: updated the list and added prizes criteria section based on meeting @abidlabs

> - HF PRO 1 month or 1 year!

Author: thanks updated, also the prizes section is still a wip as the final prizes have not been decided yet, the Gradio team put together a list in notion which I copied here 😅

- 1st place 12 months, 2nd 6m, 3rd 3m, 4th 1m HF PRO.
- Legendary prize, Lifetime HF PRO!!!

> - HF PRO badge on HF

Reviewer 3: ```suggestion
- [Hugging Face PRO subscription](https://huggingface.co/pricing) badge on Hugging Face
```

- Gradio badge on HF
- Special event badge on HF

> - A lot more swag: t-shirts, hoodies, mugs

Reviewer 3: ```suggestion
- Swag from [Hugging Face merch shop](https://huggingface.myshopify.com/): t-shirts, hoodies, mugs of your choice!
```

> - Embed the demo or highlight it

Reviewer 3: Can you elaborate on this?

Author: Thanks, yeah this section is still a WIP, will update once we finalize the prizes

Author: we had a meeting where we discussed prizes with the Gradio team, updated the section based on that, feel free to add comments/suggestions @merveenoyan

> - Time with someone on the Gradio team (unique ?)

Reviewer 3: I'd rather call it something like office-hour or 1:1 meeting.

Author: Sounds good, will update to Gradio team office-hour

Author: updated

> **Creating a Gradio demo on Hugging Face Spaces**

Author: Sounds good, thanks, I actually linked your blog for gradio + HF spaces: https://github.com/AK391/community-events/blob/61290061dc1a9e0449b3e9a00cd55ab0b2667f75/gradio-blocks/README.md?plain=1#L95 but having a seperate section on Spaces might work better here, will add it

> **Creating a Gradio demo on Hugging Face Spaces**

Author: I also have a link to the spaces docs here and a spaces section which mostly using the spaces docs, will update with the Keras sprint guide content as well thanks

https://github.com/AK391/community-events/blob/eba2a12f46f3c0f013c8c98bd4b80e9b2bdeea97/gradio-blocks/README.md?plain=1#L31

Once a model has been picked from the choices above, you can share a model in a Space using Gradio

Read more about how to add Gradio spaces [here](https://huggingface.co/blog/gradio-spaces).
 
Steps to add Gradio Spaces to the Gradio Blocks Party org

> - Create a account on Hugging Face

Reviewer 3: ```suggestion
1. Create an account on Hugging Face
```

> - Join the Gradio Blocks Party Organization by clicking Join Organization Button in the Org Page

Reviewer 3: ```suggestion
2. Join the Gradio Blocks Party Organization by clicking "Join Organization" button in the organization page
```

> - Once approved add your space using the Gradio SDK

Reviewer 3: ```suggestion
3. Once your request is approved, add your space using the Gradio SDK
```

Also it would be good to point out that people can carry already existing demos from their own profile to Gradio

# Gradio Blocks Party  🥳

![Untitled Design (1)](https://user-images.githubusercontent.com/81195143/167931943-f31e3361-55e5-4092-b0ad-5f6e85258b21.png)

_**Timeline**: May 16th, 2022 - May 31st, 2022_

---

> Happy to invite you to the Gradio Blocks Party - a community event in which we will build cool machine learning demos using the new Gradio Blocks feature. Blocks allows you to build web-based demos in a flexible way using the Gradio library. The event will take place from May 16th to 31st. We will be organizing this event on github and the huggingface discord channel.

Reviewer 4: Might be good to mention here that there are prizes up for grabs to get people's attention 🏆

Author: sounds good, Ill mention the prizes section below

Author: updated

**What is Gradio?**

Gradio is a Python library that allows you to quickly build web-based machine learning demos, data science dashboards, or other kinds of web apps, entirely in Python. These web apps can be launched from wherever you use Python (jupyter notebooks, colab notebooks, Python terminal, etc.) and shared with anyone instantly using Gradio's auto-generated share links. To learn more about Gradio see the Get Started Guide: https://gradio.app/getting_started/

Gradio can be installed via pip

```pip install gradio```

**What is Blocks?**

> `gradio.Blocks` is a low-level API that allows you to have full control over the data flows and layout of your application. You can build very complex, multi-step applications using Blocks. If you have already used gradio.Interface, you know that you can easily create fully-fledged machine learning demos with just a few lines of code. The Interface API is very convenient but in some cases may not be sufficiently flexible for your needs. For example, you might want to:

Reviewer 4: ```suggestion
`gradio.Blocks` is a low-level API that allows you to have full control over the data flows and layout of your application. You can build very complex, multi-step applications using Blocks. If you have already used `gradio.Interface`, you know that you can easily create fully-fledged machine learning demos with just a few lines of code. The Interface API is very convenient but in some cases may not be sufficiently flexible for your needs. For example, you might want to:
```

Author: thanks updated

- participant ideas, try out your own ideas

**Prizes**

- [Hugging Face PRO subscription](https://huggingface.co/pricing) 1 month for top 25 winners and 1 year for 1st place winner!
- [Hugging Face PRO subscription](https://huggingface.co/pricing) badge on Hugging Face for winners
- Gradio badge for everyone participating!
- Blocks event badge on HF for all participants!
- Swag from [Hugging Face merch shop](https://huggingface.myshopify.com/): t-shirts, hoodies, mugs of your choice for top 10 winners!
- Embedding your Gradio Blocks demo in the Gradio Blog for 1st place winner
- Gradio Team Office Hour for 1st place winner

**Prizes Criteria**

- Staff Picks
- Most liked Spaces
- Community Pick (voting)
- Most Creative Space (voting)

**Creating a Gradio demo on Hugging Face Spaces**

Once a model has been picked from the choices above or feel free to try your own idea, you can share a model in a Space using Gradio

Read more about how to add Gradio spaces [here](https://huggingface.co/blog/gradio-spaces).
 
Steps to add Gradio Spaces to the Gradio Blocks Party org
1. Create an account on Hugging Face

> 2. Join the Gradio Blocks Party Organization by clicking "Join Organization" button in the organization page

Reviewer 4: Could link https://huggingface.co/Gradio-Blocks here again, otherwise readers will have to look for it higher up in the blog post.

Author: will link thanks

Author: updated
